rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java

```diff
@@ -350,13 +350,14 @@ private NodeSet doSelectNodes(Criteria criteria, Client client) {
 
         // the nodes are selected, now mark them as busy.
         int counter = 0;
+        int usageInfoCriteriaSize = criteria.getListUsageInfo() != null ? criteria.getListUsageInfo().size() : 0;
         int listCapacity = selectedNodes.size() +
                            (selectedNodes.getExtraNodes() != null ? selectedNodes.getExtraNodes().size() : 0);
         List<String> selectedNodesUrls = new ArrayList<>(listCapacity);
         List<Map<String, String>> usageInfoList = new ArrayList<>(listCapacity);
         for (Node node : selectedNodes) {
-            if (criteria.getListUsageInfo() != null) {
-                if (criteria.getListUsageInfo().size() == 1) {
+            if (usageInfoCriteriaSize != 0) {
+                if (usageInfoCriteriaSize == 1) {
```

Comment:
The structure could be a bit more clear with less nested if:
```
if (usageInfoCriteriaSize == 1) { ...
} else if (usageInfoCriteriaSize != 0) { ...
} else { ...
}
```

Reply:
thanks, I will improve it

Reply:
else if (usageInfoCriteriaSize > 0) :-)